R/aggregate_game_stats_def.R

```diff
@@ -0,0 +1,524 @@
+      tidyselect::contains("penalty_player_") & tidyselect::ends_with("_id")
+    ) %>%
+    tidyr::pivot_longer(
+      cols = c(
+        tidyselect::contains("penalty_player_")
+      ),
+      names_to = "desc",
+      names_prefix = "penalty_",
+      values_to = "penalty_player_id",
+      values_drop_na = TRUE
+    ) %>%
+    dplyr::mutate(n = 1) %>%
+    tidyr::pivot_wider(
+      names_from = desc,
+      values_from = n,
+      values_fn = sum,
+      values_fill = 0
+    ) %>%
+    dplyr::select(
+      "season", "week",
+      "player_id" = "penalty_player_id",
+      "penalty" = "player_id"
+    )
+
+  # Touchdown stats -----------------------------------------------------------
+
+  # get defensive touchdowns
+  touchdown_df <- data %>%
+    dplyr::filter(.data$touchdown == 1) %>%
+    dplyr::filter(.data$defteam == .data$td_team) %>%
+    dplyr::group_by(
+      .data$season, .data$week,
+      "player_id" = .data$td_player_id
+    ) %>%
+    dplyr::summarise(td = sum(.data$touchdown)) %>%
+    dplyr::ungroup()
+
+  # Combine all stats -------------------------------------------------------
+
+  # combine all the stats together
+
+  player_df <- tackle_df %>%
+    dplyr::full_join(tackle_yards_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(pressure_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(int_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(safety_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(fumble_df_own, by = c("season","week","player_id")) %>%
+    dplyr::full_join(fumble_df_opp, by = c("season","week","player_id")) %>%
+    dplyr::full_join(fumble_yds_own_df, by = c("season","week","player_id")) %>%
+    dplyr::full_join(fumble_yds_opp_df, by = c("season","week","player_id")) %>%
+    dplyr::full_join(penalty_df, by = c("season","week","player_id")) %>%
+    dplyr::full_join(touchdown_df, by = c("season","week","player_id")) %>%
+    dplyr::mutate_if(is.numeric, tidyr::replace_na, 0) %>%
+    dplyr::left_join(
+      nflreadr::load_players() %>%
+        dplyr::select(
+          "player_name" = "display_name",
+          "player_id" = "gsis_id",
+          "position"
+        ),
+      by = "player_id"
+    ) %>%
+    dplyr::select(tidyselect::any_of(c(
+
+      # game information
+      "season", "week",
+
+      # id information
+      "player_id", "player_name", "team", "recent_team", "position",
+
+      # tackle stats
+      "tackles", "tackles_solo", "tackles_with_assist", "tackle_assists",
+      "tackles_for_loss", "tackles_for_loss_yards" = "tfl_yards", "forced_fumbles",
+
+      # pressure stats
+      "sacks", "sack_yards", "qb_hit",
+
+      # coverage stats
+      "int", "pass_defended", "int_yards",
+
+      # misc stats
+      "td", "fumble", "fumble_recovery_own", "fumble_recovery_yards_own",
+      "fumble_recovery_opp", "fumble_recovery_yards_opp", "safety", "penalty"
```

Comment:
Naming and order of these variables should be reviewed

Reply:
I agree. In particular, we have some weirdness where some variables are singular (e.g. `qb_hit`) and some are plural (e.g. `sacks`)